### PR TITLE
gitqlient: init at 1.1.0

### LINKED ISSUE #86702: request: GitQlient
**Project description**
GitQlient: Multi-platform Git client written with Qt. 

**Metadata**

https://github.com/francescmm/GitQlient

A non-serious start: https://github.com/colemickens/nixpkgs/commits/gitqlient

* homepage URL: https://github.com/francescmm/GitQlient
* source URL: https://github.com/francescmm/GitQlient
* license: lgpl2
* platforms: unix, linux, darwin, ...

### PR DESCRIPTION
**Pull request by colemickens**
###### Motivation for this change
Fixes: #86702.

Seems to build, launch, run okay. Going to test for a bit, maybe wait for the actual 1.1.0 release and then mark this as not-Draft.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/version-management/gitqlient/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{ stdenv, mkDerivation, fetchFromGitHub
+, qtbase, qmake
+}:
+
+with stdenv.lib;
```
Comment (prusnak) [inline pkgs/applications/version-management/gitqlient/default.nix:5]: Remove this line

File: pkgs/applications/version-management/gitqlient/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  buildInputs = [ qtbase ];
+  nativeBuildInputs = [ qmake ];
+  installFlags = [ "INSTALL_ROOT=$(out)" ];
+
+  postInstall = ''
+    mv $out/homeless-shelter/GitQlient/bin $out/bin
+  '';
+
+  meta = {
```
Comment (prusnak) [inline pkgs/applications/version-management/gitqlient/default.nix:27]:
```suggestion
  meta = with stdenv.lib; {
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -3693,6 +3693,8 @@ in
 
   gitaly = callPackage ../applications/version-management/gitlab/gitaly { };
 
+  gitqlient = qt5.callPackage ../applications/version-management/gitqlient { };
```
Comment (prusnak) [inline pkgs/top-level/all-packages.nix:3696]: It seems it is much more common to use 

```suggestion
  gitqlient = libsForQt5.callPackage ../applications/version-management/gitqlient { };
```

File: pkgs/applications/version-management/gitqlient/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{ stdenv, mkDerivation, fetchFromGitHub
```
Comment (prusnak) [inline pkgs/applications/version-management/gitqlient/default.nix:1]: Remove `stdenv`

File: pkgs/applications/version-management/gitqlient/default.nix [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  src = fetchFromGitHub {
+    owner = "francescmm";
+    repo = "GitQlient";
+    rev = "v${version}";
+    sha256 = "1x3i2w2xdvq7fknrpdcqsmf8bp9s9w8hyw2f3c3fvpyrhmm5ck28";
+    fetchSubmodules = true;
+  };
+
+  buildInputs = [ qtbase ];
+  nativeBuildInputs = [ qmake ];
```
Comment (prusnak) [inline pkgs/applications/version-management/gitqlient/default.nix:20]: Not sure if these 2 lines are required when you use `libsForQt5` in `all-packages.nix`.

File: pkgs/applications/version-management/gitqlient/default.nix [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  installFlags = [ "INSTALL_ROOT=$(out)" ];
+
+  postInstall = ''
+    mv $out/homeless-shelter/GitQlient/bin $out/bin
+  '';
+
+  meta = {
+    homepage = "https://github.com/francescmm/GitQlient";
+    description = "GitQlient: Multi-platform Git client written with Qt";
```
Comment (prusnak) [inline pkgs/applications/version-management/gitqlient/default.nix:29]:
```suggestion
    description = "Multi-platform Git client written with Qt";
```

**Review by prusnak — CHANGES_REQUESTED** [review]

File: pkgs/applications/version-management/gitqlient/default.nix (hunk 4 above)
Comment (SuperSandro2000) [inline pkgs/applications/version-management/gitqlient/default.nix:1]:
```suggestion
{ lib, mkDerivation, fetchFromGitHub
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by sikmir** [comment]
1.3.0 is out.

File: pkgs/applications/version-management/gitqlient/default.nix [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    license = licenses.lgpl2;
```
Comment (sikmir) [inline pkgs/applications/version-management/gitqlient/default.nix:30]:
```suggestion
    license = licenses.lgpl2Plus;
```

File: pkgs/applications/version-management/gitqlient/default.nix [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+mkDerivation rec {
+  pname = "GitQlient";
+  version = "1.1.0";
+
+  src = fetchFromGitHub {
+    owner = "francescmm";
+    repo = "GitQlient";
+    rev = "v${version}";
+    sha256 = "048mmwx0nv7sb2cpwl1h8wqb4q7dcnkrrmhvq08yznylgdcgkgs1";
+    fetchSubmodules = true;
```
Comment (sikmir) [inline pkgs/applications/version-management/gitqlient/default.nix:16]: There is not submodules, at least now.

File: pkgs/applications/version-management/gitqlient/default.nix (hunk 5 above)
Comment (sikmir) [reply pkgs/applications/version-management/gitqlient/default.nix:20]: The first one is not needed, but the second one is required.

Comment (sikmir) [reply pkgs/applications/version-management/gitqlient/default.nix:20]: As of 1.3.0, qtwebengine is required.

File: pkgs/applications/version-management/gitqlient/default.nix [hunk 9]
```diff
@@ -0,0 +1,34 @@
+{ stdenv, mkDerivation, fetchFromGitHub
+, qtbase, qmake
```
Comment (sikmir) [inline pkgs/applications/version-management/gitqlient/default.nix:2]:
```suggestion
, qtwebengine, qmake
```

File: pkgs/applications/version-management/gitqlient/default.nix [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+}:
+
+with stdenv.lib;
+
+mkDerivation rec {
+  pname = "GitQlient";
+  version = "1.1.0";
```
Comment (sikmir) [inline pkgs/applications/version-management/gitqlient/default.nix:9]:
```suggestion
  version = "1.3.0";
```

File: pkgs/applications/version-management/gitqlient/default.nix (hunk 8 above)
Comment (sikmir) [inline pkgs/applications/version-management/gitqlient/default.nix:16]:
```suggestion
    sha256 = "1j3mlf1yqfnm509pjvhfba754algrj94m15fxx61xkqdwkp611wz";
```

File: pkgs/applications/version-management/gitqlient/default.nix [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+  src = fetchFromGitHub {
+    owner = "francescmm";
+    repo = "GitQlient";
+    rev = "v${version}";
+    sha256 = "048mmwx0nv7sb2cpwl1h8wqb4q7dcnkrrmhvq08yznylgdcgkgs1";
+    fetchSubmodules = true;
+  };
+
+  buildInputs = [ qtbase ];
```
Comment (sikmir) [inline pkgs/applications/version-management/gitqlient/default.nix:19]:
```suggestion
  buildInputs = [ qtwebengine ];
```

File: pkgs/applications/version-management/gitqlient/default.nix [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+  nativeBuildInputs = [ qmake ];
+  installFlags = [ "INSTALL_ROOT=$(out)" ];
+
+  postInstall = ''
+    mv $out/homeless-shelter/GitQlient/bin $out/bin
+  '';
```
Comment (sikmir) [inline pkgs/applications/version-management/gitqlient/default.nix:25]:
```suggestion
```

File: pkgs/applications/version-management/gitqlient/default.nix [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+
+  meta = {
+    homepage = "https://github.com/francescmm/GitQlient";
+    description = "GitQlient: Multi-platform Git client written with Qt";
+    license = licenses.lgpl2;
+    platforms = platforms.linux;
```
Comment (sikmir) [inline pkgs/applications/version-management/gitqlient/default.nix:31]:
```suggestion
    platforms = platforms.unix;
```
It works on darwin with the following change:
```nix
  installPhase = lib.optionalString stdenv.isDarwin ''
    mkdir -p $out/Applications
    mv gitqlient.app $out/Applications
    wrapQtApp $out/Applications/gitqlient.app/Contents/MacOS/gitqlient
  '';
```

**Comment by sikmir** [comment]
@colemickens Could you please update and undraft PR?

**Comment by colemickens** [comment]
Abandoning this. There's lots of changes requested, I'm more likely to use `git-ui` in the terminal anyway. Closing for hygiene, but I'd be happy to test/review if someone wants to pick this up.

**Pull request closed without merging**